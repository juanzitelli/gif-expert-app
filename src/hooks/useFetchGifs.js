import { useState } from "react";
import { useEffect } from "react";
import { getGifs } from "../controllers/gifs-controller";

export const useFetchGifs = (category) => {
  const [gifs, setGifs] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((images) => {
      setGifs({
        data: images,
        loading: false,
      });
    });
  }, [category]);

  return gifs; 

};


