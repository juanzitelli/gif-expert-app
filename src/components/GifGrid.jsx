// Dependencies
import React from 'react'
import PropTypes from 'prop-types'

// Hooks
import { useFetchGifs } from '../hooks/useFetchGifs'
// Components
import GifGridItem from './GifGridItem'


const GifGrid = ({ category }) => {

	const { data: images, loading } = useFetchGifs(category);

	return (
		<div>
			<h3 id="category">
				{category}
			</h3>

			{loading && <p className="animate__animated animate__flash">Loading... ‚è±</p>}


			<div className="cardsContainer animate__animated animate__fadeIn animate__delay-2s">
				{images.map((image) => <GifGridItem key={image.id} {...image} />)}
			</div>
		</div>
	)
}

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
}

export default GifGrid
