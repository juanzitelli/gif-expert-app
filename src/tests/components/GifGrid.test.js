import React from "react";
import "@testing-library/jest-dom";
import GifGrid from "../../components/GifGrid";
import { shallow } from "enzyme";
import {useFetchGifs} from "../../hooks/useFetchGifs";
// import "@testing-library/jest-dom";
jest.mock("../../hooks/useFetchGifs");

describe("GifGrid tests", () => {
  test("GifGrid debe renderizarse correctamente ", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category="Hello" />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar items cuando se cargan imÃ¡genes con useFetchGif", () => {
    const test_gifs = [
      {
        id: "ABC",
        url: "https://localhost/test/cosa.jpg",
        title: "Cualquier Cosa",
      },
      {
        id: "ABC",
        url: "https://localhost/test/cosa.jpg",
        title: "Cualquier Cosa",
      },
      {
        id: "ABC",
        url: "https://localhost/test/cosa.jpg",
        title: "Cualquier Cosa",
      },
      {
        id: "ABC",
        url: "https://localhost/test/cosa.jpg",
        title: "Cualquier Cosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: test_gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category="Hello" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find("GifGridItem").length).toBe(test_gifs.length);
  });
});
