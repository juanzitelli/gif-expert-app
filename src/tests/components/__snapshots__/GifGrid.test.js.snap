// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests Debe mostrar items cuando se cargan imágenes con useFetchGif 1`] = `
<div>
  <h3
    id="category"
  >
    Hello
  </h3>
  <div
    className="cardsContainer animate__animated animate__fadeIn animate__delay-2s"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/test/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/test/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/test/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/test/cosa.jpg"
    />
  </div>
</div>
`;

exports[`GifGrid tests GifGrid debe renderizarse correctamente  1`] = `
<div>
  <h3
    id="category"
  >
    Hello
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading... ⏱
  </p>
  <div
    className="cardsContainer animate__animated animate__fadeIn animate__delay-2s"
  />
</div>
`;
